fastparse/shared/src/test/scala/fastparse/iterator/IteratorTests.scala

```diff
@@ -0,0 +1,105 @@
+package fastparse.iterator
+
+import fastparse.{ElemTypeFormatter, IteratorParserInput}
+import fastparse.all._
+import utest._
+
+import scala.collection.mutable
+
+object IteratorTests extends TestSuite {
+
+  class LoggedDropsParserInput[ElemType](data: Iterator[IndexedSeq[ElemType]])
+                                        (implicit formatter: ElemTypeFormatter[ElemType])
+    extends IteratorParserInput[ElemType](data) {
+
+    var drops = mutable.Set.empty[Int]
+
+    override def dropBuffer(index: Int): Unit = {
+      drops.add(index)
+      super.dropBuffer(index)
+    }
+  }
+
+  def toInput(string: String) = new LoggedDropsParserInput[Char](string.grouped(1).map(wrapString))
+
+  val tests = TestSuite {
+    'cuts {
+      'capturing {
+        val p = P( "a" ~/ "b" ~/ "c")
+        val capt = P( p.! ~ p.! ~ p.!)
+        val input = toInput("abcabcabc")
+        val Parsed.Success(res, i) = capt.parseInput(input)
+        assert(
+          i == 9,
+          res == ("abc", "abc", "abc"),
+          Seq(3, 6, 9).forall(input.drops.contains)
```

Comment:
For this, and subsequent tests, can we use a more exact `==` check instead of a `forall`/`contain` check? These examples are small enough we should be able to enforce *exactly* where the drops are happening.